nucypher/cli/commands/status.py

```diff
@@ -59,66 +59,42 @@ def __init__(self, provider_uri, geth, poa, registry_filepath, light, network):
         self.light = light
         self.network = network
 
-    def get_registry(self, emitter, debug):
-        try:
-            eth_node = None
-            if self.geth:
-                eth_node = get_provider_process()
-
-            # Note: For test compatibility.
-            if not BlockchainInterfaceFactory.is_interface_initialized(provider_uri=self.provider_uri):
-                BlockchainInterfaceFactory.initialize_interface(provider_uri=self.provider_uri,
-                                                                provider_process=eth_node,
-                                                                poa=self.poa,
-                                                                light=self.light,
-                                                                sync=False,
-                                                                emitter=emitter)
-
-            blockchain = BlockchainInterfaceFactory.get_interface(provider_uri=self.provider_uri)
-
-            emitter.echo(message="Reading Latest Chaindata...")
-            blockchain.connect()
-        except Exception as e:
-            if debug:
-                raise
-            click.secho(str(e), bold=True, fg='red')
-            raise click.Abort
-        if self.registry_filepath:
-            registry = LocalContractRegistry(filepath=self.registry_filepath)
-        else:
-            registry = InMemoryContractRegistry.from_latest_publication(network=self.network)
-
-        return registry
+    def setup(self, general_config) -> tuple:
+        emitter = _setup_emitter(general_config)
+        registry = get_registry(network=self.network, registry_filepath=self.registry_filepath)
+        blockchain = connect_to_blockchain(emitter=emitter, provider_uri=self.provider_uri)
+        return emitter, registry, blockchain
 
 
 group_registry_options = group_options(
     RegistryOptions,
-    provider_uri=option_provider_uri(),
     geth=option_geth,
     poa=option_poa,
     light=option_light,
     registry_filepath=option_registry_filepath,
-    network=option_network,
-    )
+    network=option_network(required=True),
+    provider_uri=option_provider_uri(required=True),
```

Comment:
I think it's interesting to have this option as `required=False`. By default, if no provider_uri is given, it will take the value of the `$NUCYPHER_PROVIDER_URI` envvar, which is very convenient as it avoids having to pass the provider to configurationless commands like `nucypher status`.

Reply:
:heavy_check_mark: